### PR TITLE
Fixing misleading error message by no network available

### PR DESCRIPTION
**Pull request by SeqLaz**
Screenshot from 2024-11-27 13-06-41

Fixing https://github.com/opengisch/qfieldsync/issues/510

### CODE REVIEW

File: qfieldsync/core/cloud_api.py [hunk 1]
```diff
@@ -688,6 +688,11 @@ def _on_avatar_download_finished(self, reply: QNetworkReply, filename: str) -> N
             self.avatar_success.emit()
 
     def get_last_login_error(self) -> str:
+        try:
+            requests.get(self.url, timeout=5)
+        except requests.ConnectionError:
+            return self.tr("No internet connection. Check your connection.")
```
Comment (suricactus) [inline qfieldsync/core/cloud_api.py:694]: Can you instead use https://doc.qt.io/qt-5/qnetworkreply.html#NetworkError-enum to detect lack of internet?

Comment (SeqLaz) [reply qfieldsync/core/cloud_api.py:694]: @suricactus I tried with the following but don't return the correct string

```python
request = QNetworkRequest(QUrl(self.url))

reply = self._nam.get(request)

if reply.error() == QNetworkReply.TemporaryNetworkFailureError:
    return self.tr("No internet connection. Check your connection.")
```

image

Comment (suricactus) [reply qfieldsync/core/cloud_api.py]: You don't have to send a new request, you already have this information. Check my commit.

**Comment by suricactus** [comment]
image

**Comment by SeqLaz** [comment]
> image

Great!..... Really Thank you @suricactus!

**Pull request merged**